packages/gitbook/src/components/PageAside/PageAside.tsx

```diff
@@ -100,135 +116,159 @@ export async function PageAside(props: {
                 'page-api-block:xl:max-2xl:rounded-md',
                 'page-api-block:xl:max-2xl:h-auto',
                 'page-api-block:xl:max-2xl:my-8',
-
-                topOffset,
+                'page-api-block:p-2',
             )}
         >
+            {page.layout.outline ? (
+                <>
+                    <div
+                        className={tcls(
+                            'hidden',
+                            'page-api-block:xl:max-2xl:flex',
+                            'text-xs',
+                            'tracking-wide',
+                            'font-semibold',
+                            'uppercase',
+
+                            'flex-row',
+                            'items-center',
+                            'gap-2',
+                        )}
+                    >
+                        <Icon icon="block-quote" className={tcls('size-3')} />
+                        {t(language, 'on_this_page')}
+                        <Icon
+                            icon="chevron-down"
+                            className={tcls(
+                                'size-3',
+                                'opacity-6',
+                                'ml-auto',
+                                'page-api-block:xl:max-2xl:group-hover/aside:hidden',
+                            )}
+                        />
+                    </div>
+                    <div
+                        className={tcls(
+                            'overflow-auto',
+                            'overflow-x-visible',
```

Comment:
Potential conflict there for the y overflow, means the resulting behavior is unpredictable, better to set them separately:
```suggestion
                            'overflow-y-auto',
                            'overflow-x-visible',
```